### PR TITLE
adding Explorer Time Matrix userscript

### PR DESCRIPTION
**Pull request by SasSam**
It calculates and shows all the times (treasure and adventure searches) for all the explorers you have based on their skills. You can check the normal and the PFB times as well.

**Please note**: if you have loads of explorers (like 150+), the modal window will load slowly, so be patient!

The **greyed-out times** show that the explorer doesn't have that skill or the user is not yet on the required level for that search.

Screenshot examples:

**Normal times (no PFB) - Treasure searches**
image

**PFB times - Treasure searches**
image

**Normal times (no PFB) - Adventure searches**
image

**PFB times - Adventure searches**
image

### CODE REVIEW

File: userscripts/user_exp_time_matrix.js [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+ *
+ * You can find me on Discord (sas_sam).
+ *
+ * If you like my work, you can support me: https://ko-fi.com/sas_sam
+ *
+ */
+var ExplorerTimeMatrix_PFB_RB_STATUS = false;
+var ExplorerTimeMatrix_ADV_RB_STATUS = false;
+
+function logger(message) {
```
Comment (fedorovvl) [inline userscripts/user_exp_time_matrix.js:20]: client already have function "debug" with same logic.. use it

File: userscripts/user_exp_time_matrix.js [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+      game.gi.mZoneBuffManager.isBuffRunning("MultiplierBuffZone2_PremiumFriendBuff1Day")
+    };
+
+    this.getAvailableExplorers = function () {
+      var _this = this;
+      var availableExplorers = [];
+
+      game
+        .getSpecialists()
+        .sort(0)
```
Comment (fedorovvl) [inline userscripts/user_exp_time_matrix.js:203]: players can choose sort method in settings.. try to use sort(specNameSorter)
specNameSorter is function from common.js

File: userscripts/user_exp_time_matrix.js [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+
+          availableExplorers.push(item);
+        });
+
+        this.totalExplorers = availableExplorers.length;
+
+        return availableExplorers;
+    };
+
+    this.getTaskInfo = function (taskId, subTaskId) {
```
Comment (fedorovvl) [inline userscripts/user_exp_time_matrix.js:221]: u can use getTaskInfo from 4-specialists.js.. not necessary to duplicate logic

File: userscripts/user_exp_time_matrix.js (hunk 1 above)
Comment (SasSam) [reply userscripts/user_exp_time_matrix.js]: removed

File: userscripts/user_exp_time_matrix.js (hunk 3 above)
Comment (SasSam) [reply userscripts/user_exp_time_matrix.js]: Function removed and replaced with the global reference from `4-specialists.js`.

**Pull request merged**